app/models/course/achievement.rb

```diff
@@ -11,4 +11,13 @@ class Course::Achievement < ActiveRecord::Base
   def set_defaults
     self.weight ||= 10
   end
+
+  # Returns all achievements that this achievement may have as a condition.
+  #
+  # @return [Array<Course::Achievement>]
+  def valid_achievement_conditions
+    course.achievements - [self] -
+      specific_conditions.select { |c| c.is_a?(Course::Condition::Achievement) }.
+        map(&:achievement)
+  end
```

Comment:
this still is the wrong solution right? You cannot join from a polymorphic relation, but you can join to a polymorphic relation. Imagine the SQL that you need to construct.